src/arch/io/network.cc

```diff
@@ -621,6 +621,430 @@ void linux_tcp_conn_t::on_event(int /* events */) {
     event_watcher->stop_watching_for_errors();
 }
 
+/* tls_conn_wrapper_t wraps a TLS connection and safely releases it in
+exceptional cases if it is not reclaimed before execution leaves its scope. */
+class tls_conn_wrapper_t {
+public:
+    tls_conn_wrapper_t(SSL *_conn) {
```

Comment:
This constructor should have an `explicit` keyword
```C++
explicit tls_conn_wrapper_t(SSL *_conn) {
```

(I'll probably have a bunch of minor stylistic things like this. I hope you don't mind. These things are fairly arbitrary, but you can find our style guide at https://github.com/rethinkdb/rethinkdb/blob/daniel_coding_style/STYLE.md )